### DISCUSSION TITLE
Use the DefaultGenericBehaviour DSL with the aster interface

### DISCUSSION THREAD

**Discussion by mhpierre** [thelfer/tfel#394] [Q&A]
When trying to compile a law written with the DefaultGenericBehaviour DSL with the aster interface, I get the following:

`the aster interface only supports small and finite strain behaviours and cohesive zone models`

Does this mean that this DSL cannot be used with Code Aster ? I would like to use multi physics laws in code aster and I think the generic DSL is the only way I can implement them in MFront.

Thanks in advance,
Maxime

**Comment by thelfer** [comment]
Hi @LordKthulhu,

Unfortunately, no, you can't use generalized multi-physics behaviors in `code_aster`.

There is however one exception: point-wise models. What are you planning, maybe a workaround is possible ?

**Reply by mhpierre** [reply]
I have a coupled thermo-poro-mechanical model with plasticity working in MFront written with the` ImplicitGenericBehaviour` DSL, and I would like to use it in `code_aster`.  So I would need temperature and pressure as external variables in addition to deformation. I guess for simplicity, let's say that I want to use this thermo-mechanical model from this demo: [mgis_fenics_monolithic_transient_thermoelasticity](https://thelfer.github.io/mgis/web/mgis_fenics_monolithic_transient_thermoelasticity.html). Could there be a workaround to use this behaviour in `code_aster` ?

**Reply by thelfer** [reply]
Simple anwser: no you can't. However, you can use MFront "standard" mechanical behaviours in THM in code_aster, but you don't have any hand on how suction and heat flux are computed. @SRaude, any help ?

**Reply by SRaude** [reply]
Hi @LordKthulhu and @thelfer, 

Actually, we are working on introducing the capillary pressure as an external state variable in __MFront__ (for BBM constitutive model for example).

I think this function is already available if you have an updated development version of __Code_Aster__ installed via gitlab https://gitlab.com/codeaster.

You can get the values of __PCAP__  (at time +) and __dPCAP__ (increment) and include them in your constitutive equations. The temperature is already available as an external state variable.

If you use your model with __CREA_LIB_MFRONT__, i think you have nothing to do to be able to use it in TH(H)M conditions.

**Reply by SRaude** [reply]
In the last updated version of __Code_Aster__, if you are using your __MFront__ model as an official one; i.e. you have to create and update Python files to introduce your model in __Code_Aster__ (not only use it as an external library), the interface between __MFront__ and __Code_Aster__ is now __MGIS__.

You can grep __MohrCoulombAS__, __Iwan__, or __NLH_CSRM__ in src files to take a look on how to introduce your __MFront__ model as an official one (what are the Python files you have to update/create).

**Reply by mhpierre** [reply]
Thank you very much @thelfer and @SRaude. I will try with the dev version of code_aster then.
